src/recipe.cpp

```diff
@@ -444,28 +440,67 @@ std::string recipe::required_skills_string( const Character *c, bool print_skill
     } );
 }
 
-std::string recipe::required_skills_string( const Character *c ) const
-{
-    return required_skills_string( c, false );
-}
-
-std::string recipe::required_skills_string() const
+// Format a std::pair<skill_id, int> for the basecamp bulletin board.
+// skill colored white with difficulty in parenthesis.
+template<typename _FIter>
+std::string required_skills_as_string( _FIter first, _FIter last )
 {
-    if( required_skills.empty() ) {
-        if( difficulty == 0 ) {
-            return _( "<color_cyan>none</color>" );
-        } else {
-            return string_format( "<color_white>%s: %d</color>", skill_used.obj().name(),
-                                  difficulty );
-        }
+    if( first == last ) {
+        return _( "<color_cyan>none</color>" );
     }
-    return enumerate_as_string( required_skills.begin(), required_skills.end(),
+
+    return enumerate_as_string( first, last,
     [&]( const std::pair<skill_id, int> &skill ) {
-        return string_format( "<color_white>%s: %d</color>", skill.first.obj().name(),
+        return string_format( "<color_white>%s (%d)</color>", skill.first.obj().name(),
                               skill.second );
     } );
 }
 
+std::string recipe::primary_skill_string( const Character *c, bool print_skill_level ) const
+{
+    std::list< std::pair<skill_id, int> > skillList;
+
+    if( !skill_used.is_null() ) {
+        skillList.push_back( std::pair<skill_id, int>( skill_used, difficulty ) );
+    }
+
+    return required_skills_as_string( skillList.begin(), skillList.end(), c, print_skill_level );
+}
+
+std::string recipe::required_skills_string( const Character *c, bool include_primary_skill,
+        bool print_skill_level ) const
+{
+    // There is no primary skill used or it shouldn't be included then we can just use the required_skills directly.
+    if( skill_used.is_null() || !include_primary_skill ) {
+        return required_skills_as_string( required_skills.begin(), required_skills.end(), c,
+                                          print_skill_level );
+    }
+
+    std::list< std::pair<skill_id, int> > skillList;
+    skillList.push_back( std::pair<skill_id, int>( skill_used, difficulty ) );
+    for( const std::pair<skill_id, int> &skill : required_skills ) {
```

Comment:
Consider using `std::copy` with a `back_inserter` rather than a `for` loop.  Similarly below.